source/state_representation/include/state_representation/geometry/Shape.hpp

```diff
@@ -139,29 +158,33 @@ inline const CartesianTwist& Shape::get_center_twist() const {
 }
 
 inline void Shape::set_center_state(const CartesianState& state) {
-  this->set_name(state.get_name());
+  if (state.is_empty()) {
```

Comment:
I don't think that the name of the center state should be linked to the shape name